### PR TITLE
Remove generator dependency of expose.go

### LINKED ISSUE #93100: Remove generators from kubectl commands
As @soltysh suggests:
kubectl command are going to get rid of the dependency to the generators, and ultimately remove them entirely.
The main goal is to simplyfy the use of kubectl command.
This is already done in `kubectl run`, see https://github.com/kubernetes/kubernetes/pull/87077 and https://github.com/kubernetes/kubernetes/pull/68132.

Example PR that removes generators : https://github.com/kubernetes/kubernetes/pull/91127

Related files: 
- [x]  `pkg/cmd/create/create_rolebinding.go` (@zhouya0, #92956)
- [x]  `pkg/cmd/create/create_clusterrolebinding.go` (@aubm , #91127)
- [x]  `pkg/cmd/create/create_deployments.go` (@aubm, #90676)
- [ ]   `pkg/cmd/create/create_namespaces.go` (@CaiyuhangC, https://github.com/kubernetes/kubernetes/pull/96556)
- [x]   `pkg/cmd/create/create_priorityclass.go` (@aubm)
- [x]   ` pkg/cmd/create/create_pdb.go` (@sachinkumarsingh092, https://github.com/kubernetes/kubernetes/pull/97429)
- [x]   `pkg/cmd/create/create_quota.go` (@SaiHarshaK, https://github.com/kubernetes/kubernetes/pull/95265)
- [x]   `pkg/cmd/create/create_configmap.go` (@lauchokyip https://github.com/kubernetes/kubernetes/pull/97981)
- [ ]   `pkg/cmd/create/create_secret.go` (@lauchokyip, https://github.com/kubernetes/kubernetes/pull/98614)
- [x]   `pkg/cmd/create/create_service.go` (@Git-Jiro, https://github.com/kubernetes/kubernetes/pull/95256)
- [x]   `pkg/cmd/create/create_serviceaccount.go` (@ramnar, https://github.com/kubernetes/kubernetes/pull/96492)
 
- [x]   `pkg/cmd/autoscale/autoscale.go` (@SataQiu, #92998)
- [ ]   `pkg/cmd/expose/expose.go` (@lauchokyip)
- [ ]    `pkg/cmd/run/run.go` (@soltysh )

/cc @aubm @rahulkrishnanfs @SataQiu

### PR DESCRIPTION
**Pull request by lauchokyip**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Remove the generators dependency from `kubectl expose`

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/93100

#### Does this PR introduce a user-facing change?

```release-note
Remove deprecated generator and container-port flags
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (lauchokyip) [inline staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:37]: Seem like version 1 generator has been deprecated https://github.com/kubernetes/kubectl/blob/b1d9ef54819c3666364d2a0ea49cf3f7faf7454f/pkg/generate/versioned/generator.go#L40

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (lauchokyip) [inline staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:150]: Seem like version 1 generator has been deprecated https://github.com/kubernetes/kubectl/blob/b1d9ef54819c3666364d2a0ea49cf3f7faf7454f/pkg/generate/versioned/generator.go#L40

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (lauchokyip) [inline staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:158]: This has been deprecated for 5 years already, so I removed this

**Comment by lauchokyip** [comment]
@eddiezane  Maybe you can help out for this one too?

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:150]: Yeah, it was removed earlier this year in https://github.com/kubernetes/kubernetes/pull/99733

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:158]: Mention this flag and `service/v1` generator removal in release notes, explicitly.

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:284]: Reading through code, I see we never validated that, I'd probably add a check for empty or `None` value for this one.

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:309]: I'd prefer we switch all internal errors to just return error, so `fmt.Errorf("couldn't find protocol via introspection: %v", err)` in the rest of the file as well. This allows consumers of the command to wrap the error and respond accordingly. Whereas `cmdutil.UsafeErrorf` adds unnecessarily kubectl specific information, ie. `See 'kubectl expose -h' for help and examples`, etc, which are already nicely handled in the cobra's `Run` command.

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:310]: `service/v2` has two required values: `default-name` and `selector` see https://github.com/kubernetes/kubectl/blob/1bf219da9bdd8c03758195f5b43f1ddfbe31c3e7/pkg/generate/versioned/service.go#L55-L57 make sure to have appropriate `Validate` method to reflect that.

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:390]: As mentioned before move this to `Validate` method which you don't have, yet.

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:415]: Move this to `Validate`

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:435]: Probably parts of this validation could be in `Validate` but I'll leave it up to you to figure this one out.

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:481]: Same here, wrt validation.

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:509]: Move to `Validate`

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:393]: Probably worth moving that to `Validate`

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:400]: Similarly, move to `Validate`.

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose_test.go
Comment (soltysh) [inline staging/src/k8s.io/kubectl/pkg/cmd/expose/expose_test.go:862]: Very good extensive tests :+1:

**Review by soltysh — CHANGES_REQUESTED** [review]

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:158]: Added

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:309]: Fixed

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:310]: both of that was done in `createService()` , but will move it to `Validate()`

Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:310]: `default-name` cannot be moved to `Validate` because if user does not provide a name it will call the API server and get from the `info` object.

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:415]: `default-name` cannot be moved to `Validate` because if user does not provide a name it will call the API server and get from the `info` object which mean user does not have to provide a name beforehand. 
https://github.com/kubernetes/kubectl/blob/1bf219da9bdd8c03758195f5b43f1ddfbe31c3e7/pkg/cmd/expose/expose.go#L254

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:390]: See comment above

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:393]: See comment above

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:284]: If you look at the test cases, https://github.com/kubernetes/kubectl/blob/1bf219da9bdd8c03758195f5b43f1ddfbe31c3e7/pkg/generate/versioned/service_test.go, cluster-ip can be any value, `""`, `10.10.10.10` or `None`

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (lauchokyip) [reply staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:400]: See comment above

**Comment by lauchokyip** [comment]
@soltysh , I don't think anything can be changed anymore. I have already thought of moving to `Validate` before submitting the PR but `expose.go` is different before validation is actually done inside `RunExpose`

**Comment by dims** [comment]
looks like @lauchokyip has responses for all the comments, back to approver!

/assign @soltysh

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Comment by eddiezane** [comment]
[Flake](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/106824/pull-kubernetes-conformance-kind-ga-only-parallel/1508949144452993024) is https://github.com/kubernetes/kubernetes/issues/108891

/retest

**Pull request merged**
